testing/integration/upgrade_standalone_same_commit_test.go

```diff
@@ -0,0 +1,381 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/dev-tools/mage"
+	v1 "github.com/elastic/elastic-agent/pkg/api/v1"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/testcontext"
+	"github.com/elastic/elastic-agent/pkg/version"
+	"github.com/elastic/elastic-agent/testing/upgradetest"
+	agtversion "github.com/elastic/elastic-agent/version"
+)
+
+func TestStandaloneUpgradeSameCommit(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Upgrade,
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+
+	// parse the version we are testing
+	currentVersion, err := version.ParseVersion(define.Version())
+	require.NoError(t, err)
+
+	// 8.13.0-SNAPSHOT is the minimum version we need for testing upgrading with the same hash
+	if currentVersion.Less(*upgradetest.Version_8_13_0_SNAPSHOT) {
+		t.Skipf("Minimum version for running this test is %q, current version: %q", *upgradetest.Version_8_13_0_SNAPSHOT, currentVersion)
+	}
+
+	unprivilegedAvailable := true
+	if runtime.GOOS != define.Linux {
+		// only available on Linux at the moment
+		unprivilegedAvailable = false
+	}
+	// This is probably redundant: see the skip statement above
+	if unprivilegedAvailable && (currentVersion.Less(*upgradetest.Version_8_13_0) || currentVersion.Less(*upgradetest.Version_8_13_0)) {
```

Comment:
```suggestion
	if unprivilegedAvailable && currentVersion.Less(*upgradetest.Version_8_13_0) {
```

Both sides of this `||` are the same expression, since the start version and end version here are the same I suspect you can remove one of them.

Reply:
Oups I forgot to remove it after I changed the second version to 8.13.0 (we have a similar check for the privileged/unprivileged switch that you can find in other tests).
Will remove.